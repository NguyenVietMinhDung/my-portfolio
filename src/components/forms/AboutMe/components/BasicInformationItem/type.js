export type Props = {
  iconClassName: string,
  name: string,
  value: string,
};
