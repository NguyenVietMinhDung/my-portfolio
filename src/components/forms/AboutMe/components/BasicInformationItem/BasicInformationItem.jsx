// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './BasicInformationItem.scss';
import type { Props } from './type';

const BasicInformationItem = (props: Props) => {
  const { iconClassName, name, value } = props;
  return (
    <li
      key={name}
      styleName="basic-info-item"
    >
      <i
        styleName="icon"
        className={iconClassName}
      />
      <div styleName="text">
        <span>{name}:</span>
        <span>&nbsp;{value}</span>
      </div>
    </li>
  );
};

export default memo<Props>(CSSModules(BasicInformationItem, styles));
