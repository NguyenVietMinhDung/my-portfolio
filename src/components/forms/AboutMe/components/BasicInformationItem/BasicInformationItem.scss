@import '../../../../../styles/abstracts/variables';

.basic-info-item {
	display: flex;

	margin-top: 5px;
	align-items: center;
	flex-flow: row wrap;

	.icon {
		color: $violet-color;
	}

	.text {
		padding-left: 10px;

		span:first-child {
			color: $black-color;
		}
	}
}
