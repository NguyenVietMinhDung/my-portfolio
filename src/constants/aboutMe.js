export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
