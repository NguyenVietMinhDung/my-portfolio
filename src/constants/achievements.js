export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
