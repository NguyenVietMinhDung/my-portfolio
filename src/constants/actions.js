export const ACTIVATE_NAVIGATION_ITEM = 'ACTIVATE_NAVIGATION_ITEM';
export const OPEN_NAVIGATION_MENU = 'OPEN_NAVIGATION_MENU';
