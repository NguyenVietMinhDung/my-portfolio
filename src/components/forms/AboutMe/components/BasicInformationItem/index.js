import BasicInformationItem from './BasicInformationItem';

export default BasicInformationItem;
