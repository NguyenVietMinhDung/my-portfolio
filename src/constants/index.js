import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
