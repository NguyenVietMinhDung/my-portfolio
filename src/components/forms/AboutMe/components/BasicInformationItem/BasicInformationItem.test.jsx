import React from 'react';
import { shallow } from 'enzyme';
import { basicInfo as items } from '../../../../../constants/aboutMe';
import BasicInformationItem from './BasicInformationItem';

const defaultProps = items[0];

describe('<BasicInformationItem />', () => {
  test('renders correctly', () => {
    const component = shallow(<BasicInformationItem {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
