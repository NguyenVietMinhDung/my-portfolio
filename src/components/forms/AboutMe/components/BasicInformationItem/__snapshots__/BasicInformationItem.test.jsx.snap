// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicInformationItem /> renders correctly 1`] = `
<li
  className="basic-info-item"
  key="Date of birth"
>
  <i
    className="lni-calendar icon"
  />
  <div
    className="text"
  >
    <span>
      Date of birth
      :
    </span>
    <span>
      Â 
      16 January 1995
    </span>
  </div>
</li>
`;
